File: app/schemas/user_log.py
# app/schemas/user_log.py
from pydantic import BaseModel
from datetime import datetime

class UserLogRead(BaseModel):
    id: int
    method: str
    path: str
    status_code: int
    duration: float
    timestamp: datetime
    ip: str | None = None
    user_agent: str | None = None

    class Config:
        from_attributes = True

File: express-app/schemas/user_log.js
// express-app/schemas/user_log.js
const Joi = require('joi');

const userLogReadSchema = Joi.object({
  id: Joi.number().integer().required(),
  method: Joi.string().required(),
  path: Joi.string().required(),
  status_code: Joi.number().integer().required(),
  duration: Joi.number().required(),
  timestamp: Joi.date().iso().required(),
  ip: Joi.string().allow(null).optional(),
  user_agent: Joi.string().allow(null).optional(),
});

module.exports = {
  userLogReadSchema,
};